### ISSUE TITLE
Use ZIP instead of TGZ with Windows

### ISSUE THREAD

**Issue by cup** [racket/racket#3002]
I see here:

https://download.racket-lang.org/releases/7.5

That Windows has Installer and Tarball options. To my knowledge Windows doesnt
have any built in tools to extract TGZ files. This isnt a huge deal as you can
get 7-Zip or similar to extract, but it might make sense to distribute ZIP for
the Windows releases.

**Comment by cup** [comment]
Looks like I was mistaken

https://devblogs.microsoft.com/commandline/tar-and-curl-come-to-windows

**Comment by mflatt** [comment]
I agree that ".zip" would be more convenient. But last I knew, ".zip" extractors tended not to preserve timestamps from an archive, which creates trouble for an unpacked Racket installation. Extraction of ".tar" or ".tgz" files tend to preserve timestamps. So, that's why we have provided ".tgz" archives. Then again, if my impression of ".zip" extraction is wrong or outdated, ".zip" files would be better.

**Comment by cup** [comment]
@mflatt I just tested and that doesnt seem to be the case. Using Windows 7 at least, I am seeing that timestamps are preserved using the built in extraction of Windows Explorer.

If you have different experience let me know

**Comment by sorawee** [comment]
It seems that on Windows, the default zip extractor indeed occasionally modifies the timestamp. It is done for security reason. See https://devblogs.microsoft.com/oldnewthing/20110504-00/?p=10743

**Comment by cup** [comment]
> which creates trouble for an unpacked Racket installation

What kind of trouble does it create? I have dealt with ZIP distributions of many programs over the years and dont recall this ever being a problem. Just want to make sure we are not dealing with a [Chesterton fence](//en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence)

**Comment by mflatt** [comment]
The trouble is that Racket (by default) compares the timestamp on a ".rkt" to the timestamp on a corresponding compiled ".zo" file to decide whether to use the compiled file. If the relative order of timestamps changes, then loading programs like DrRacket can be very slow.

**Comment by jackfirth** [comment]
Could the zo files be tweaked to embed the timestamp directly in their content, and could Racket use that as a fallback when the regular timestamp looks like its in the past?

(I have no idea if that's a good idea or not, I'm just curious.)

**Comment by mflatt** [comment]
I'm not sure how embedding a timestamp would help, since the problem happens with a comparison timestamp of the ".zo" file with the corresponding ".rkt" file, and the timestamp is just as likely to go wrong on the ".rkt" file.

Embedding source and dependency hashes, similar to ".dep" files, could be a way to go. Checking those would take time, but that might be worthwhile if the alternative is to start compiling from source. I'm not sure about that, either.
